reframe/frontend/cli.py

```diff
@@ -375,46 +375,27 @@ def main():
         # Locate and load checks
         try:
             checks_found = loader.load_all()
-        except OSError as e:
+        except OSError as oe:
             raise ReframeError from e
 
         # Filter checks by name
-        checks_matched = filter(
-            lambda c:
-            c if c.name not in options.exclude_names else None,
-            checks_found
-        )
+        checks_matched = filter(filters.have_not_name(options.exclude_names), checks_found)
 
         if options.names:
-            checks_matched = filter(
-                lambda c: c if c.name in options.names else None,
-                checks_matched
-            )
+            checks_matched = filter(filters.have_name(options.names), checks_matched)
 
         # Filter checks by tags
         user_tags = set(options.tags)
```

Comment:
There is no need for this one any more. The `have_tags` converts its argument to a set already.

Reply:
Removed.